### PR TITLE
Nessie error details: Add `ContentKey` to related errors

### PR DESCRIPTION
**Pull request by snazy**
Add a `ContentKeyErrorDetails` to all `ContentKey` related exceptions as an error-details implementation, so that the `ContentKey` can be extracted on the client side.

Also move all error-details to `org.projectnessie.model.error`.

### CODE REVIEW

**Comment by snazy** [comment]
@dimas-b @adutra WDYT about adding `ContentKey` to content related errors/exceptions?

File: api/model/src/main/java/org/projectnessie/error/NessieError.java [hunk 1]
```diff
@@ -69,6 +68,16 @@ default ErrorCode getErrorCode() {
   @JsonInclude(Include.NON_NULL)
   NessieErrorDetails getErrorDetails();
 
+  default <T extends NessieErrorDetails> T getErrorDetailsAsOrNull(Class<T> type) {
+    NessieErrorDetails details = getErrorDetails();
+    if (details != null && type.isAssignableFrom(details.getClass())) {
+      @SuppressWarnings({"UnnecessaryLocalVariable", "unchecked"})
+      T casted = (T) details;
```
Comment (dimas-b) [inline api/model/src/main/java/org/projectnessie/error/NessieError.java:75]: nit: `type.cast(details)` and avoid `@SuppressWarnings`

File: integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseCreateReferenceExec.scala [hunk 2]
```diff
@@ -46,7 +47,7 @@ abstract class BaseCreateReferenceExec(
           NessieUtils.getCurrentRef(api, currentCatalog, catalog)
         } catch {
           case e: NessieNotFoundException =>
-            throw new NessieNotFoundException(
+            throw new NessieReferenceNotFoundException(
```
Comment (dimas-b) [inline integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseCreateReferenceExec.scala:50]: Should we have `NessieReferenceNotFoundException` as a `case` type too (line 49)?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -742,7 +742,7 @@ jobs:
       ICEBERG_MAIN_REPOSITORY: apache/iceberg
       ICEBERG_MAIN_BRANCH: master
       ICEBERG_PATCH_REPOSITORY: snazy/iceberg
-      ICEBERG_PATCH_BRANCH: iceberg-nesqueit
+      ICEBERG_PATCH_BRANCH: iceberg-nesqueit-err
```
Comment (dimas-b) [inline .github/workflows/ci.yml:745]: Do we still need a custom iceberg branch?

**Review by dimas-b — COMMENTED** [review]
Looks like a good idea to me :)

File: .github/workflows/ci.yml (hunk 3 above)
Comment (snazy) [reply .github/workflows/ci.yml:745]: Yea :(

File: integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseCreateReferenceExec.scala (hunk 2 above)
Comment (snazy) [reply integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseCreateReferenceExec.scala:50]: Not sure actually. Quite some time ago. But yes: all the Spark things need a lot of love.

File: servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/BaseTestNessieApi.java [hunk 4]
```diff
@@ -1269,7 +1280,11 @@ public void namespaces() throws Exception {
     if (isV2()) {
       soft.assertThatThrownBy(
               () -> api().deleteNamespace().refName(mainName).namespace(namespace2).delete())
-          .isInstanceOf(NessieNamespaceNotEmptyException.class);
+          .isInstanceOf(NessieNamespaceNotEmptyException.class)
+          .asInstanceOf(type(NessieNamespaceNotEmptyException.class))
```
Comment (dimas-b) [inline servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/BaseTestNessieApi.java:1284]: nit: (mostly wondering) do we need both `isInstanceOf` and `asInstanceOf`?

Comment (snazy) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/BaseTestNessieApi.java:1284]: think so

**Pull request merged**